### PR TITLE
Add read

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: scaleway/helpers.go [hunk 1]
```diff
@@ -285,6 +286,24 @@ func getZone(d terraformResourceData, meta *Meta) (utils.Zone, error) {
 	return utils.Zone(""), fmt.Errorf("could not detect region")
 }
 
+// extractInstanceAPICreateHelper returns a new instance API and the zone for a Create request
+func extractInstanceAPICreateHelper(d *schema.ResourceData, m interface{}) (*instance.API, utils.Zone, error) {
```
Comment (QuentinBrosse) [inline scaleway/helpers.go]: Why `Helper` suffix ?

File: scaleway/resource_compute_instance_server.go [hunk 2]
```diff
@@ -90,7 +89,7 @@ func resourceScalewayComputeInstanceServer() *schema.Resource {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Default:     "started",
-				Description: "the server action (poweron, poweroff)",
+				Description: "the state the server should be (started,stopped,standby)",
```
Comment (QuentinBrosse) [inline scaleway/resource_compute_instance_server.go]: the state **of** the server

File: scaleway/resource_compute_instance_server.go [hunk 3]
```diff
@@ -178,5 +173,129 @@ func resourceScalewayComputeInstanceServerCreate(d *schema.ResourceData, m inter
...
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("name", response.Server.Name)
+	d.Set("image", response.Server.Image.ID)
```
Comment (QuentinBrosse) [inline scaleway/resource_compute_instance_server.go:191]: We have to check if it is a UUID of an image name

File: scaleway/instance_helpers.go [hunk 4]
```diff
@@ -16,10 +16,10 @@ func getInstanceAPIWithZone(d *schema.ResourceData, m interface{}) (*instance.AP
 }
 
 // getInstanceAPIWithZoneAndID returns an instance API with zone and ID extracted from the state
-func getInstanceAPIWithZoneAndID(d *schema.ResourceData, m interface{}) (*instance.API, utils.Zone, string, error) {
+func getInstanceAPIWithZoneAndID(m interface{}, id string) (*instance.API, utils.Zone, string, error) {
```
Comment (QuentinBrosse) [inline scaleway/instance_helpers.go]: Rename file `compute_instance_helpers` ?

File: scaleway/resource_compute_instance_server.go [hunk 5]
```diff
@@ -261,7 +263,7 @@ func resourceScalewayComputeInstanceServerUpdate(d *schema.ResourceData, m inter
 }
 
 func resourceScalewayComputeInstanceServerDelete(d *schema.ResourceData, m interface{}) error {
```
Comment (QuentinBrosse) [inline scaleway/resource_compute_instance_server.go:265]: So we agree that `ServerActionTerminate ` don't remove any volumes?

File: scaleway/resource_compute_instance_server_test.go [hunk 6]
```diff
@@ -1,12 +1,109 @@
...
+			},
+		},
+	})
+}
+
+func testAccCheckScalewayComputeInstanceServerExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
```
Comment (QuentinBrosse) [inline scaleway/resource_compute_instance_server_test.go]: "resource not found" ?

File: scaleway/resource_compute_instance_server.go (hunk 5 above)
Comment (kindermoumoute) [reply scaleway/resource_compute_instance_server.go:265]: It does remove the root volume!

**Pull request merged**